core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitorMessageParser.scala

```diff
@@ -0,0 +1,60 @@
+package org.broadinstitute.dsde.rawls.monitor
+
+import org.broadinstitute.dsde.rawls.dataaccess.SlickDataSource
+import org.broadinstitute.dsde.rawls.google.GooglePubSubDAO.PubSubMessage
+import org.broadinstitute.dsde.rawls.model.{RawlsUserEmail, WorkspaceName}
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+
+class AvroUpsertMonitorMessageParser(message: PubSubMessage, dataSource: SlickDataSource)(implicit
+  executionContext: ExecutionContext
+) {
+
+  def parse: Future[AvroUpsertAttributes] = {
+    val workspaceNamespace = "workspaceNamespace"
+    val workspaceName = "workspaceName"
+    val userEmail = "userEmail"
+    val jobId = "jobId"
+    val upsertFile = "upsertFile"
+    val isUpsert = "isUpsert"
+
+    val workspaceId = "workspaceId"
+
+    def attributeNotFoundException(attribute: String): Nothing = throw new Exception(
+      s"unable to parse message - attribute $attribute not found in ${message.attributes}"
```

Comment:
`$attribute` doesn't also need {} around it?

Reply:
heh, nope - if it's just a single variable, no {} needed; if it's an expression (like including the .) then they are needed.